merkletree/proof_test.go

```diff
@@ -53,6 +53,13 @@ func TestVerifyProof(t *testing.T) {
 
 	// proof of inclusion
 	proof := m.Get(index3)
+	if proof.Leaf.Value() == nil {
+		t.Fatal("Expect returned leaf's value is not nil")
+	}
+	// ensure this is a proof of inclusion
+	if !bytes.Equal(proof.LookupIndex, proof.Leaf.Index()) {
```

Comment:
I think I would feel more comfortable checking that ``proof.LookupIndex == privateIndex(name) && LookupIndex == proof.Leaf.Index()``. It only becomes a proof of inclusion once both of these conditions hold.